templates/keepalived.conf.j2

```diff
@@ -271,12 +273,10 @@ virtual_server {{ vserver.ip }} {{ vserver.port }} {
     }
     {% endfor %}
     {% endif %}
-    {% if rserver.dns_check is defined %}
+    {% if rserver.dns_checks is defined %}
     {% for dns_check in rserver.dns_checks %}
     DNS_CHECK {
-      {% if dns_check.connect_ip is defined and dns_check.connect_ip %}
       connect_ip {{ dns_check.connect_ip }}
```

Comment:
i've made the connect_ip option mandatory even if it is optional for keepalived.
just to make sure that there is a value in that dict to prevent ansible dict errors.